app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,123 @@
+class OrdersController < ApplicationController
+  before_action :current_user
+
+  before_action :check_user_id, only: [:index, :show, :ship]
+
+
+  def index
+    user_id = params[:user_id]
+    @user = User.find(user_id)
+    @orders = []
+    @user.products.each do |product|
+      product.order_items.each do |oi|
+        if ['pending','paid','cancelled','complete'].include? params[:sort]
+          if oi.order.status == params[:sort]
+            if @orders.include? oi.order
+              next
```

Comment:
If both of these conditionals need to be true to move on, you can combine them on one line using `&&`.

`if oi.order.status == params[:sort] && if @orders.include? oi.order`